### PR TITLE
[PM-3050] Add sync on unlock, logout when account is deleted

### PR DESCRIPTION
**Pull request by quexten**
## Type of change

```
- [x] Bug fix
- [ ] New feature development
- [ ] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [ ] Other
```

## Objective

Notifications for logout or account changes are only processed while an account is unlocked. This means that if an account change occurs while locked, or the account gets deleted while locked, the client (browser/desktop) does not know about this, even after unlocking. This leads to items being desynced, and if the account was deleted, the client is in a broken state.

This PR fixes that, by, upon unlock checking if a full sync is required. Since fullSync checks the account revision timestamp, the "expensive" full sync only happens when needed, and thus prevents a desynced state. Further, the account revision timestamp is negative if (and only if) the account was deleted. In this case we log the client out. (We might want to change this on the server side to give a proper error noting that the account does not exist).

## Code changes

- **file.ext:** Description of what was changed and why

## Before you submit

- Please add **unit tests** where it makes sense to do so (encouraged but not required)
- If this change requires a **documentation update** - notify the documentation team
- If this change has particular **deployment requirements** - notify the DevOps team
- Ensure that all UI additions follow [WCAG AA requirements](https://contributing.bitwarden.com/contributing/accessibility/)

### CODE REVIEW

**Comment by quexten** [comment]
Before:


https://github.com/bitwarden/clients/assets/11866552/71037634-5542-43bc-92d4-0f01d66dd77d

**Comment by quexten** [comment]
After:

https://github.com/bitwarden/clients/assets/11866552/7cd6df94-925e-47ec-89eb-05cbb20e5589

File: libs/common/src/vault/services/sync/sync.service.ts [hunk 1]
```diff
@@ -145,6 +145,13 @@ export class SyncService extends CoreSyncService {
     }
 
     const response = await this.apiService.getAccountRevisionDate();
+    if (response < 0) {
```
Comment (gbubemismith) [inline libs/common/src/vault/services/sync/sync.service.ts:148]: suggestion: I think we can avoid the nested ifs 

```suggestion
    if (response < 0 && this.logoutCallback) {
          // Account was deleted, log out now
          await this.logoutCallback(false);
    }
```

**Review by gbubemismith — CHANGES_REQUESTED** [review]
This is looking good, @quexten, I just added one suggestion, and I think some unit tests are failing

**Comment by quexten** [comment]
I should stop usirg rebase...

**Comment by quexten** [comment]
Forgot to inject state service in the unit tests, should be fixed now.

**Review by JaredSnider-Bitwarden — APPROVED** [review]
Nice work on this!

**Pull request merged**